### PR TITLE
Catalogue of My Things

### PR DESCRIPTION
**Pull request by danielufeli**
This project was made by our team, including @david-lafontant , @denscholar and @danielufeli .

In this project, we created a console app that will help us to keep records of different types of things we won: books, music albums, movies, and games. Everything will be based on the UML class diagram presented to us in the project requirement document. The data will be stored in JSON files but you will also prepare a database with the structure of a table analogical to our program's class structure.

- Each class from the UML class diagram is created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes are implemented.
- All properties visible in the diagram are defined and set up in the constructor method.
- We managed the project utilizing a Kanban board.

### CODE REVIEW

File: classes/item.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    author.add_item(self)
+  end
+
+  def add_label(label)
+    label.add_item(self)
+  end
+
+  def add_genre(genre)
+    genre.add_item(self)
+  end
```
Comment (medaminedev66) [inline classes/item.rb:26]: [OPTIONAL] The associations (1-to-many relationships) should be implemented correctly. Item class should have one Genre, Author, and Label. You have defined instant class variables (@author, @label, and @genre) but you haven't used them in add methods. Check out the example below to learn more about that:
```ruby
class Item
    attr_reader :genre
    attr_reader :author
    attr_reader :label

    ...

    def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
    end

    def author=(author)
      @author = author
      author.items.push(self) unless author.items.include?(self)
    end

    def label=(label)
      @label = label
      label.items.push(self) unless label.items.include?(self)
    end

    ...
  end
```

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @danielufeli, @david-lafontant, and @denscholar 😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- You have implemented all the requirements, great job 👍
- The repo looks professional ✔️
- Your code organization was really good ✔️
- There are no linter errors ✔️
- You have a descriptive PR ✔️

## Required Changes ♻️
Following the project requirements:
- [x] all the team members should appear in the video and talk about the project.
- [x] The maximum length of the video should be 5 minutes.
Checkout the requirements bellow 👇:
image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/item.rb (hunk 1 above)
Comment (david-lafontant) [reply classes/item.rb:26]: Optional change impmented

**Review by elmar8287 — APPROVED** [review]
Hi, 

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**